test/minikube/minikube_sm_test.go

```diff
@@ -0,0 +1,291 @@
+// +build long
+
+/*
+ * Tests to verify operation of SM startup.
+ * Separate so that changes to SM startup in general - and the SM startup scipt(s) in particular
+ * can be tested in isolation.
+ */
+
+package minikube
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/nuodb/nuodb-helm-charts/test/testlib"
+	"github.com/stretchr/testify/assert"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/random"
+)
+
+func TestKubernetesStartSM(t *testing.T) {
```

Comment:
please dont add tests/code that never worked!

Reply:
In my defence, I did say on slack that I was going to perform a cleanup to this PR prior to re-requesting peer review.

This test is particularly difficult in an environment where there is only ever 1 SM.
Causing an error in an SM causes the database to shut down if there is only 1 SM - and it turns out there are so many ways to get to a state where the code tries to clean up, and the database is in some conflicting state - such as NOT_RUNNING, or similar.


I _will_ invest another 2-3 hours trying to make this test work by making the nested tests independent, so there is no need to heal the database.

Otherwise, I will remove this test.
I have a copy on disk, and I can try getting it to work as a separate exercise.

Reply:
oh. So I am reviewing this too early! Sorry!